### DISCUSSION TITLE
FullTextSearch + Prisma + PostgreSQL NOT WORKING

### DISCUSSION THREAD

**Discussion by Vounts** [prisma/prisma#13847] [Q&A]
I'm using PostgreSQL 14 and the latest Prisma

**Steps to Reproduce:**
1. Enabled the flag

**previewFeatures = ["fullTextSearch"]**

2. **Run npx prisma generate**
3. Did not generate any search field
4. Error when calling
5. THROWN
Error: 
```
Invalid `prisma.items.findMany()` invocation:

{
  where: {
    body: {
    ~~~~
      search: 'cat'
    }
  }
}

Unknown arg `body` in where.body for type itemsWhereInput

```
Code that thrown the error
```
 const items = await prisma.items.findMany({
      where: {
        body: {
          search: "cat",
        },
      },
    });

```

**Comment by janpio** [comment]
What does your schema look like?

**Reply by vladi-stevanovic** [reply]
👋 Hello @Vounts  I'm going through pending GitHub Discussions to check if you need any help. Is this issue still open?
Please let us know if we can investigate it further or if you've already solved it, we'd love to know how!

It would be super useful for other users facing this problem! 🙌

**Comment by abbylow** [comment]
Hi @janpio, I have a similar issue when I use FullTextSearch + Prisma + CockroachDB. 

I modified my schema to add fullTextSearch preview feature, and I found that `Searching on String fields without a full text index not supported` in CockroachDB. 
Thus, I added index for Thread's title `@@index([title], name: "thread_full_text_index")` and Post's content `@@index([content], name: "post_full_text_index", type: Gin)`. 

I ran `npx prisma generate` and `npx prisma migrate dev` (I even changed to use a brand new db).  

However, when I tried to use `search` in my findMany query, it always throw error as follows: 

### findMany query
```
await prisma.thread.findMany({
    where: {
      groupId,
      type: 'public',
      title: {
        search: searchTerm
      }
    },
    orderBy: { updatedAt: 'desc' },
    skip: THREADS_PER_PAGE * page,
    take: THREADS_PER_PAGE + 1,
    include: {
      posts: {
        where: {
          parentId: null
        }
      }
    }
  }) 
```

### Error thrown from using the query above
```
error - PrismaClientValidationError: 
Invalid `prisma.thread.findMany()` invocation:

{
  where: {
    groupId: '0x3b425aae906f7e3abdbf3d12ce424dd92902bc83972b071c298259a5298ea6bf',
    type: 'public',
    title: {
      search: 'dfdsf'
      ~~~~~~
    }
  },
  orderBy: {
    updatedAt: 'desc'
  },
  skip: 8,
  take: 9,
  include: {
    posts: {
      where: {
        parentId: null
      }
    }
  }
}

Unknown arg `search` in where.title.search for type StringFilter. Did you mean `select`? Available args:
type StringFilter {
  equals?: String
  in?: List<String>
  notIn?: List<String>
  lt?: String
  lte?: String
  gt?: String
  gte?: String
  contains?: String
  startsWith?: String
  endsWith?: String
  mode?: QueryMode
  not?: String | NestedStringFilter
}
```

## Schema
```
generator client {
  provider = "prisma-client-js"
  previewFeatures = ["fullTextSearch"]
}

datasource db {
  provider = "cockroachdb"
  url      = env("DATABASE_URL")
}

model Group {
  id         String   @id // email domain
  users      User[]
  merkleRoot String
  createdAt  DateTime @default(now())
  updatedAt  DateTime @updatedAt
  threads    Thread[]
  @@unique ([id, merkleRoot])
}

model Thread {
  id        String   @id
  groupId   String
  group     Group    @relation(fields: [groupId], references: [id])
  title     String
  posts     Post[]
  type      String
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
  @@index([title], name: "thread_full_text_index")
}

model Post {
  id        String   @id
  content   Json
  authorId  String // authorId is a hash(user identity nullfier hash, thread id)
  parentId  String? // can be post id or null (for the op post)
  parent    Post?    @relation("PostRelation", fields: [parentId], references: [id])
  threadId  String
  thread    Thread   @relation(fields: [threadId], references: [id])
  depth     Int
  children  Post[]   @relation("PostRelation")
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt

  @@index([content], name: "post_full_text_index", type: Gin)
}

```

**Reply by abbylow** [reply]
Does Prisma Client actually support full text search for CockroachDB? 

I was following the [PostgreSQL example](https://www.prisma.io/docs/concepts/components/prisma-client/full-text-search) to setup the full text search feature. Then I only realized that CockroachDB might be not supported as it is not fully compatible with PostgreSQL.
>The Prisma Client supports full-text search for PostgreSQL databases in versions 2.30.0 and later, and MySQL databases in versions 3.8.0 and later.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
